src/test/java/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterDefinitionTest.java

```diff
@@ -589,4 +591,29 @@ public void testCliBuild() throws Exception {
         assertNull(b.getExactRun(new Combination(axes, "value2")));
         assertNotNull(b.getExactRun(new Combination(axes, "value3")));
     }
+
+    @Test
+    public void testBuildWithParameters() throws Exception {
+        AxisList axes = new AxisList(new TextAxis("axis1", "value1", "value2", "value3"));
+        MatrixProject p = j.createMatrixProject();
+        p.setAxes(axes);
+        p.addProperty(new ParametersDefinitionProperty(
+            new MatrixCombinationsParameterDefinition("combinations", "", "")
+        ));
+
+        WebClient wc = j.createWebClient();
+        wc.getPage(p, String.format(
+            "/buildWithParameters?combinations=%s",
+            URLEncoder.encode("axis1 in ['value1', 'value3']", "UTF-8")

```

Comment:
> I guess that it's groovy script ?

A combination filter.
Please refer the help text of multi-configuration projects.

> That would have been nice to have compatibility :/

Not planned.
WebAPI and CLI supports are since 1.1.0.
The WebAPI behavior in 1.0.9 was not designed nor documented, and cannot be supported.